### DISCUSSION TITLE
Is there a solution for large data exchange between cores?

### DISCUSSION THREAD

**Discussion by shellyfengfeng** [OpenAMP/open-amp#536] [Q&A]
Since the RPMSG buffer size is fixed, if large data(not fixed size) are needed to exchange between two cores,  maybe RPMSG couldn't support the scenario, is there other solution to solve it?
Thanks.

**Comment by arnopo** [comment]
Similar discussion is starting on the linux mailing list: https://www.spinics.net/lists/linux-remoteproc/msg17100.html

RPMSG currently does not support dynamic buffer allocation. One solution is to use indirect buffer, where rpmsg is used to provide the address and size of allocated buffers.
Another solution would be to use VirtIO which allows for adding buffers in vring descriptors. Work is starting in OpenAMP to enable the direct use of the VirtIO protocol, but it is not yet well supported for coprocessor management.

**Reply by shellyfengfeng** [reply]
Using indirect buffer means using shared memory where service push data into directly and rpmsg pass the pointer of these special memory to the remote.
So this solution need to use special shared memory management strategy.
